main.nf

```diff
@@ -788,286 +993,574 @@ process convertBam {
+        .dump(tag: "AR Bypass")
+        .into { ch_adapterremoval_for_fastqc_after_clipping; ch_adapterremoval_for_lanemerge; } 
 } else {
     ch_fastp_for_skipadapterremoval
-        .into { ch_adapterremoval_for_fastqc_after_clipping; ch_adapterremoval_for_skipmap; ch_adapteremoval_for_bwa; ch_adapteremoval_for_cm; ch_adapteremoval_for_bwamem;  } 
+        .into { ch_adapterremoval_for_fastqc_after_clipping; ch_adapterremoval_for_lanemerge; } 
+}
+
+// Lane merging for libraries sequenced over multiple lanes (e.g. NextSeq)
+ch_branched_for_lanemerge = ch_adapterremoval_for_lanemerge
+  .groupTuple(by: [0,1,3,4,5,6])
+  .map {
+    it ->
+      def samplename = it[0]
+      def libraryid  = it[1]
+      def lane = it[2]
+      def seqtype = it[3]
+      def organism = it[4]
+      def strandedness = it[5]
+      def udg = it[6]
+      def r1 = it[7]
+      def r2 = it[8]
+
+      [ samplename, libraryid, lane, seqtype, organism, strandedness, udg, r1, r2 ]
+
+  }
+  .dump(tag: "LaneMerge Bypass")
+  .branch {
+    skip_merge: it[7].size() == 1 // Can skip merging if only single lanes
+    merge_me: it[7].size() > 1
+  }
+
+ch_branched_for_lanemerge_skipme = ch_branched_for_lanemerge.skip_merge
+  .map{
+    it -> 
+        def samplename = it[0]
+        def libraryid  = it[1]
+        def lane = it[2]
+        def seqtype = it[3]
+        def organism = it[4]
+        def strandedness = it[5]
+        def udg = it[6]
+        def r1 = it[7][0]
+        def r2 = it[8][0]
+
+        [ samplename, libraryid, lane, seqtype, organism, strandedness, udg, r1, r2 ]
+  }
+  .dump(tag: "LaneMerge Reconfigure")
+
+
+ch_branched_for_lanemerge_ready = ch_branched_for_lanemerge.merge_me
+  .map{
+      it -> 
+        def samplename = it[0]
+        def libraryid  = it[1]
+        def lane = it[2]
+        def seqtype = it[3]
+        def organism = it[4]
+        def strandedness = it[5]
+        def udg = it[6]
+        def r1 = it[7]
+
+        // find and remove duplicate dummies to prevent file collision error
+        def r2 = it[8]*.toString()
+        r2.removeAll{ it == "$baseDir/assets/nf-core_eager_dummy.txt" }
+
+        [ samplename, libraryid, lane, seqtype, organism, strandedness, udg, r1, r2 ]
+  }
+
+process lanemerge {
+  label 'sc_tiny'
+  tag "${libraryid}"
+  publishDir "${params.outdir}/lanemerging", mode: params.publish_dir_mode
+
+  input:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, path(r1), path(r2) from ch_branched_for_lanemerge_ready
+
+  output:
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, path("*_R1_lanemerged.fq.gz") into ch_lanemerge_for_mapping_r1
+  tuple samplename, libraryid, lane, seqtype, organism, strandedness, udg, path("*_R2_lanemerged.fq.gz") optional true into ch_lanemerge_for_mapping_r2
+
+  script:
+  if ( seqtype == 'PE' && ( params.skip_collapse || params.skip_adapterremoval ) ){
+  lane = 0
```

Comment:
you never use that specific value and define `lane` again in the later parts.

Reply:
This is a 'dummy' value to ensure the sample names remain in a consistent pattern for certain downstream processes. It also helps people tracking when they have performed lane merging (unfortunately aDNA people do wierd and wonderful sequencing strategies which this traceback can help sometimes....)

Reply:
but is it ever actually used? Any outcome of the following code has a `def lane=` line, no?

Reply:
Nope, actually you're right. I misremembered, I apparantly did remove the usage of it downstream.

This is a valid point to simplify code/reduce the code size, however this will require a  _lot_ of changes in many steps, that I am nervous to do so close to a stable release :sweat_smile:, as it would increase risk of minor mistakes that might break something, and it's currently not doing any harm.

I will make an issue to remove it in a future release.